Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go

```diff
@@ -534,10 +534,20 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 		}()
 	}
 	wg.Wait()
+
 	select {
 	case err := <-errs:
 		return nil, err
 	default:
+		anyLabels := listOptions.LabelSelector == nil || listOptions.LabelSelector.Empty()
+		anyFields := listOptions.FieldSelector == nil || listOptions.FieldSelector.Empty()
+
+		if anyLabels && anyFields && api.NamespaceValue(ctx) != "" {
+			// Try to delete the directory, which will only succeed if the directory is empty.
```

Comment:
Ahh!  You are right.  I mistakenly just looked at the if block for graceful and missed the outer-block.  Need to see who is configuring with `ReturnDeletedObject` as true, and I am out of time to do that now.  Shockingly, I am nervous about delete code paths.